src/System.IO.Pipes/src/System/IO/Pipes/NamedPipeClientStream.cs

```diff
@@ -84,6 +86,15 @@ public NamedPipeClientStream(String serverName, String pipeName, PipeDirection d
             {
                 throw new ArgumentOutOfRangeException(nameof(inheritability), SR.ArgumentOutOfRange_HandleInheritabilityNoneOrInheritable);
             }
+            if ((options & PipeOptions.CurrentUserOnly) != 0)
+            {
+                _isCurrentUserOnly = true;
+
+                // We need to remove this flag from options because it is not a valid flag for windows PInvoke to create a pipe.
+                options &= ~PipeOptions.CurrentUserOnly;
+            }
+
+            Debug.Assert((options & PipeOptions.CurrentUserOnly) == 0);
```

Comment:
Nit, I think this assert is a bit pointless because it is obviously satisfied a few lines above.

Reply:
Oh I see @pjanotti suggested it. .. fine ..